cdap-data-fabric/src/main/java/co/cask/cdap/data2/increment/hbase/IncrementHandlerState.java

```diff
@@ -0,0 +1,171 @@
+    String familyAsString = Bytes.toString(familyName);
+    byte[] transactionalConfig = familyValues.get(Bytes.toBytes(IncrementHandlerState.PROPERTY_TRANSACTIONAL));
+    boolean txnl = transactionalConfig == null || !"false".equals(Bytes.toString(transactionalConfig));
+    LOG.info("Family " + familyAsString + " is transactional: " + txnl);
+    if (txnl) {
+      txnlFamilies.add(familyName);
+    }
+
+    // check for TTL configuration
+    byte[] columnTTL = familyValues.get(Bytes.toBytes(TxConstants.PROPERTY_TTL));
+    long ttl = 0;
+    if (columnTTL != null) {
+      try {
+        String stringTTL = Bytes.toString(columnTTL);
+        ttl = Long.parseLong(stringTTL);
+        LOG.info("Family " + familyAsString + " has TTL of " + ttl);
+      } catch (NumberFormatException nfe) {
+        LOG.warn("Invalid TTL value configured for column family " + familyAsString +
+            ", value = " + Bytes.toStringBinary(columnTTL));
+      }
+    }
+    ttlByFamily.put(familyName, ttl);
+
+    // get the transaction state cache as soon as we have a transactional family
+    if (!txnlFamilies.isEmpty() && cache == null) {
+      Supplier<TransactionStateCache> cacheSupplier = getTransactionStateCacheSupplier(tableName, conf);
+      this.cache = cacheSupplier.get();
+    }
+  }
+
+  public boolean containsTransactionalFamily(Set<byte[]> familyNames) {
+    // we assume that if any of the column families written to are transactional, the entire write is transactional
+    for (byte[] key : familyNames) {
+      if (txnlFamilies.contains(key)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Returns a unique timestamp for the current operation.
+   *
+   * @return a new timestamp guaranteed to be unique within the scope of usage
+   */
+  public synchronized long getUniqueTimestamp() {
+    return timeOracle.getUniqueTimestamp();
+  }
+
+  /**
+   * Returns the upper bound beyond which we can compact any increment deltas into a new sum.
+   * @param columnFamily the column family name
+   * @return the newest timestamp beyond which can compact delta increments
+   */
+  public long getCompactionBound(byte[] columnFamily) {
+    if (txnlFamilies.contains(columnFamily)) {
+      TransactionSnapshot snapshot = cache.getLatestState();
+      // if tx snapshot is not available, used "0" as upper bound to avoid trashing in-progress tx
+      return snapshot != null ? snapshot.getVisibilityUpperBound() : 0;
+    } else {
+      return Long.MAX_VALUE;
+    }
+  }
+
+  /**
+   * Returns the time-to-live (in milliseconds) for the given column family, transformed into the same precision
+   * used in assigning unique timestamps.
+   *
+   * @param familyName the column family name
+   * @return the time-to-live value
+   */
+  public long getFamilyTTL(byte[] familyName) {
+    Long configuredTTL = ttlByFamily.get(familyName);
+    return configuredTTL == null ? -1 : configuredTTL * TxConstants.MAX_TX_PER_MS;
```

Comment:
I may be wrong - but we may tell that ttl is in seconds. Or it is presented to user as sec, but internally in ms. Confused - I remember asked shankar to clean it, but not sure what it is right now. You may want to doublecheck